connectors/src/api/create_connector.ts

```diff
@@ -145,13 +146,26 @@ const _createConnectorAPIHandler = async (
     }
 
     if (connectorRes.isErr()) {
-      return apiError(req, res, {
-        status_code: 500,
-        api_error: {
-          type: "internal_server_error",
-          message: connectorRes.error.message,
-        },
-      });
+      switch (connectorRes.error.classification) {
+        case "INTERNAL":
```

Comment:
NIT: EXPECTED / UNEXPECTED? 

Or discussed IRL: If we return Err it's a 400, and 500 if we throw